### PR TITLE
refactor: Change return type of IsMine to isminetype && move wallet files to wallet directory

### LINKED ISSUE #1648: listsinceblock and listtransactions are not showing send transactions
The rpc api commands **listsinceblock** and **listtransactions** are not showing transactions with category **send**. The transactions with other categories are listed correctly.

The issue is happening using the version **4.0.6.0-Leisure** installed with the package gridcoinresearchd_4.0.6.0_stretch-1_amd64.deb at **Debian Linux**

### PR DESCRIPTION
**Pull request by div72**
This PR changes the return type of `IsMine` to `isminetype`; just like in bitcoin
Also moves some wallet files to `src/wallet`
Fixes #1648
Compiles fine, runs with no apparent issues. `listtransactions` and `listsinceblock` returns sent transactions without setting watch only to true.

### CODE REVIEW

File: src/script.cpp [hunk 1]
```diff
@@ -1605,10 +1616,27 @@ bool IsMine(const CKeyStore &keystore, const CScript& scriptPubKey)
...
         vector<valtype> keys(vSolutions.begin()+1, vSolutions.begin()+vSolutions.size()-1);
-        return HaveKeys(keys, keystore) == keys.size();
+        if (HaveKeys(keys, keystore) == keys.size()) {
+            ret = std::max(ret, IsMineResult::SPENDABLE);
+        }
+        break;
     }
     }
-    return false;
+    // TODO: Watch-only addresses
```
Comment (cyrossignol) [inline src/script.cpp:1625]: I think we don't store watch-only addresses at all yet, right?

**Review by cyrossignol — APPROVED** [review]
Looks good. I like the explicit comparison to `ISMINE_NO`. Bitcoin seems to rely on the implicit conversion of `isminetype` to `bool`. We will need to decide which style to keep when we port these sections of the Bitcoin code. It may be less work to follow Bitcoin's convention. What do you think?

File: src/script.cpp (hunk 1 above)
Comment (jamescowens) [reply src/script.cpp:1625]: Correct, but we should eventually support it...

**Review by jamescowens — APPROVED** [review]

**Comment by jamescowens** [comment]
@div72, @cyrossignol is getting a segfault ... we may have to revert this if we can't find an easy solution.

**Comment by cyrossignol** [comment]
```
Thread 1 "gridcoinresearc" received signal SIGSEGV, Segmentation fault.
std::vector<unsigned char, std::allocator<unsigned char> >::vector (this=0x7ffea13303c0,
    __x=<error reading variable: Cannot access memory at address 0x8>) at /usr/include/c++/9/bits/stl_vector.h:552
552             _Alloc_traits::_S_select_on_copy(__x._M_get_Tp_allocator()))
(gdb) bt
#0  std::vector<unsigned char, std::allocator<unsigned char> >::vector (this=0x7ffea13303c0, __x=<error reading variable: Cannot access memory at address 0x8>) at /usr/include/c++/9/bits/stl_vector.h:552
#1  0x0000564497def6b5 in IsMineInner (keystore=..., scriptPubKey=...) at ../../src/script.cpp:1592
#2  0x0000564497defa4d in IsMine (keystore=..., scriptPubKey=...) at ../../src/script.cpp:1631
#3  0x0000564497e6bf1a in CWallet::IsMine (txout=..., this=0x5644993ac640) at ../../src/wallet/wallet.h:337
#4  CWallet::FixSpentCoins (this=<optimized out>, nMismatchFound=@0x7ffea133089c: 0, nBalanceInQuestion=@0x7ffea13308a0: 0, fCheckOnly=fCheckOnly@entry=false) at ../../src/wallet/wallet.cpp:2453
#5  0x0000564497c48b86 in AppInit2 (threads=...) at ../../src/init.cpp:1184
#6  0x0000564497c2e3c9 in AppInit (argc=3, argv=0x7ffea1331988) at /usr/include/c++/9/ext/atomicity.h:96
#7  0x0000564497c1c556 in main (argc=3, argv=0x7ffea1331988) at ../../src/gridcoinresearchd.cpp:190
```

In `IsMineInner()`, there's an out-of-bounds access to the `vSolutions` vector filled by `Solver()`. Looking into it...

**Comment by cyrossignol** [comment]
See #1736 for a potential solution.

**Pull request merged**